### DISCUSSION TITLE
Ask a question about Zaraz issues using cloudflare.

### DISCUSSION THREAD

**Discussion by wyfang** [gildas-lormeau/SingleFile#1464] [Q&A]
I've been using SingleFile and I've been using Zaraz, I would save an html using SingleFile and then write some js code in Zaraz to control the html, like adding buttons to all html. But I realized that the buttons are not displayed in the current SingleFile saved page, while I visit the SingleFile saved page from a year ago and it will be there. I'm wondering if there is some setting in the condition that is causing this? The external js can't control to the content in the html?

**Comment by gildas-lormeau** [comment]
Your issue might be related to the presence of the [CSP](https://developer.mozilla.org/en-US/docs/Web/HTTP/CSP) in the HTML via the `<meta>` tag (see below) which is present in the `<head>` tag of the saved page. 

```html
<meta http-equiv="content-security-policy" content="default-src 'none'; font-src 'self' data:; img-src 'self' data:; style-src 'unsafe-inline'; media-src 'self' data:; script-src 'unsafe-inline' data:; object-src 'self' data:; frame-src 'self' data:;">
```

Maybe the `script-src` policy blocks the script you're trying to inject and execute. You can set the hidden option `insertMetaCSP` to `false` to verify if it's the cause of the problem, see https://github.com/gildas-lormeau/SingleFile/wiki/Hidden-options.
